### PR TITLE
LieAlgebras: Fix combinatorics tests

### PR DESCRIPTION
**Pull request by lgoettgens**
Adds tests for edge cases of wrappers around Hecke functions, that have been fixed in thofma/Hecke.jl#1040.
Furthermore, does no longer do `using submodule` in testing and thus messing up the namespaces.

### CODE REVIEW

File: experimental/LieAlgebras/test/Combinatorics-test.jl [hunk 1]
```diff
@@ -1,5 +1,7 @@
 @testset "LieAlgebras.Combinatorics" begin
   @testset "combinations" begin
+    combinations = Oscar.LieAlgebras.combinations
```
Comment (fingolfin) [inline experimental/LieAlgebras/test/Combinatorics-test.jl:3]: Shouldn't this be 
```suggestion
    import Oscar.LieAlgebras: combinations
```
Possibly at the top of this file? Same for the other similar ones

Comment (lgoettgens) [reply experimental/LieAlgebras/test/Combinatorics-test.jl:3]: In my understanding, `import` would pull it into the current namespace. What I was trying to achieve here, is only to be able to use the short form in this one test set without polluting the global namespace.
I see the possibility for the success of the test ci being dependent on the order of the test files run, if every file pulls something more into the global namespace, and possibly using something non-exported that has only been imported by a previous test.

If this, in your opinion, is no problem, or if there is a cleaner way to do it, I am happy to change it.

Comment (thofma) [reply experimental/LieAlgebras/test/Combinatorics-test.jl:3]: Yes, this is the right way to do it. Alternatively use a `let` around everything.

Comment (fingolfin) [reply experimental/LieAlgebras/test/Combinatorics-test.jl:3]: or put the tests into a module of their own?

Comment (thofma) [reply experimental/LieAlgebras/test/Combinatorics-test.jl:3]: I learned to use `let` from the julia tests (if it is not already inside a `@testset` block). There it is used all over the place to avoid polluting the namespace. Not sure if it makes the use of `let` better.

Comment (thofma) [reply experimental/LieAlgebras/test/Combinatorics-test.jl:3]: P.S.: I thought `@testset` introduces automatically a local scope block.

Comment (lgoettgens) [reply experimental/LieAlgebras/test/Combinatorics-test.jl:3]: I am a bit confused. What would be the/your preferred way to change this into?

Comment (fingolfin) [reply experimental/LieAlgebras/test/Combinatorics-test.jl:3]: Either use `import` or `let` or put everything into a module -- but don't do `combinations = Oscar.LieAlgebras.combinations`.

@thofma suggests using `let`, which seems to fit with the goal you want to achieve; I'd be happy with that as well

Comment (thofma) [reply experimental/LieAlgebras/test/Combinatorics-test.jl:3]: I think `@testset` introduces a local scope, so the `combinations = ...` should be fine? @fingolfin can you elaborate what the problem is?

Comment (lgoettgens) [reply experimental/LieAlgebras/test/Combinatorics-test.jl:3]: Yes, indeed there is a local scope. The following fails with an UndefVarError in line 6.
```julia
@testset "Some tests" begin
  x = 3
  @test x == 3
end

@test x == 3
```

**Review by fingolfin — APPROVED** [review]
